app/resources/game_content.py

```diff
@@ -0,0 +1,189 @@
+import math
+from typing import Tuple, List
+
+from app import m
+from models.hardware import Hardware
+from models.service import Service
+from models.workload import Workload
+from vars import hardware, resolve_ram_type, resolve_gpu_type
+
+
+def check_exists(user: str, elements: dict) -> Tuple[bool, dict]:
+    response: dict = m.contact_microservice("inventory", ["inventory", "list"], {"owner": user})
+
+    names: List[str] = [x["element_name"] for x in response["elements"]]
+
+    if elements["cpu"] not in names:
+        return False, {"error": "you_dont_own_such_cpu"}
+    if elements["motherboard"] not in names:
+        return False, {"error": "you_dont_own_such_motherboard"}
+    if elements["gpu"] not in names:
+        return False, {"error": "you_dont_own_such_gpu"}
+    for ram in elements["ram"]:
+        if ram not in names:
+            return False, {"error": "you_dont_own_such_ram"}
+        else:
+            names.remove(ram)
+    for disk in elements["disk"]:
+        if disk not in names:
+            return False, {"error": "you_dont_own_such_disk"}
+        else:
+            names.remove(disk)
+
+    return True, {}
+
+
+def delete(user: str, elements: dict):
+    m.contact_microservice("inventory", ["inventory", "delete_by_name"], {"owner": user, "item_name": elements["cpu"]})
+    m.contact_microservice(
+        "inventory", ["inventory", "delete_by_name"], {"owner": user, "item_name": elements["motherboard"]}
+    )
+    m.contact_microservice("inventory", ["inventory", "delete_by_name"], {"owner": user, "item_name": elements["gpu"]})
+    for ram in elements["ram"]:
+        m.contact_microservice("inventory", ["inventory", "delete_by_name"], {"owner": user, "item_name": ram})
+    for disk in elements["disk"]:
+        m.contact_microservice("inventory", ["inventory", "delete_by_name"], {"owner": user, "item_name": disk})
+
+
+def check_element_existence(elements: dict) -> Tuple[bool, dict]:
+    if elements["cpu"] not in hardware["cpu"]:
+        return False, {"error": "element_cpu_not_found"}
+    if elements["gpu"] not in hardware["gpu"]:
+        return False, {"error": "element_gpu_not_found"}
+    if elements["motherboard"] not in hardware["mainboards"]:
+        return False, {"error": "element_motherboard_not_found"}
+    for disk in elements["disk"]:
+        if disk not in hardware["disk"]:
+            return False, {"error": "element_disk_not_found"}
+    for ram in elements["ram"]:
+        if ram not in hardware["ram"]:
+            return False, {"error": "element_ram_not_found"}
+
+    return True, {}
+
+
+def check_compatible(elements: dict) -> Tuple[bool, dict]:
+    exists, message = check_element_existence(elements)
+    if not exists:
+        return False, message
+
+    cpu: str = elements["cpu"]
+    motherboard: str = elements["motherboard"]
+    ram: List[str] = elements["ram"]
+    gpu: str = elements["gpu"]
```

Comment:
variable `gpu` is not used

Reply:
Lets see ... but I can not find any attribute in vars.py where we could check for compatibility

Reply:
@DevMimas Can we except an adjusted config file in the near future ?

Reply:
maybe later